addons/sale_project/tests/test_sale_project.py

```diff
@@ -652,3 +652,98 @@ def test_quick_create_sol(self):
         sale_line = self.env['sale.order.line'].browse(sale_line_id)
         self.assertEqual(sale_line.product_id, product_service, 'The created SOL should use the right product.')
         self.assertTrue(product_service.name in sale_line_name, 'The created SOL should use the full name of the product and not just what was typed.')
+
+    def test_sale_order_with_project_task_from_multi_companies(self):
+        SaleOrder = self.env["sale.order"]
+        Company = self.env["res.company"]
+        Product = self.env["product.product"]
+        Project = self.env["project.project"]
+        Partner = self.env["res.partner"]
+
+        uom_hour = self.env.ref("uom.product_uom_hour")
+        will_smith = Partner.create({"name": "Will Smith"})
+        multi_company_project = Project.create({
+            "name": "Multi Company Project",
+            "company_id": None,
+            "allow_billable": True,
+        })
+
+        company_a = Company.create({
+            "name": "Company A"
+        })
+        service_product_a = Product.with_company(company_a).create({
+            "name": "Task Creating Product",
+            "standard_price": 30,
+            "list_price": 90,
+            "type": "service",
+            "service_tracking": "task_global_project",
+            "invoice_policy": "order",
+            "uom_id": uom_hour.id,
+            "uom_po_id": uom_hour.id,
+            "project_id": multi_company_project.id,
+        })
+        sale_order_a_1 = SaleOrder.with_company(company_a).create({
+            "partner_id": will_smith.id,
+            "order_line": [Command.create({
+                "product_id": service_product_a.id,
+                "product_uom_qty": 10,
+            })]
+        })
+        sale_order_a_2 = SaleOrder.with_company(company_a).create({
+            "partner_id": will_smith.id,
+            "order_line": [Command.create({
+                "product_id": service_product_a.id,
+                "product_uom_qty": 10,
+            })]
+        })
```

Comment:
Can be done in batch as well. 🙂 

```suggestion
        sale_order_a_1, sale_order_a_2 = SaleOrder.with_company(company_a).create([
            {
                "partner_id": will_smith.id,
                "order_line": [
                    Command.create({
                        "product_id": service_product_a.id,
                        "product_uom_qty": 10,
                    }),
                ],
            }, {
                "partner_id": will_smith.id,
                "order_line": [
                    Command.create({
                        "product_id": service_product_a.id,
                        "product_uom_qty": 10,
                    }),
                ],
            },
        ])
```